createFolders.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+PROJECT_NAME=data-and-algorithms
+DIRECTORY=${PWD}
+BASE_DIRECTORY=$(echo "$DIRECTORY" | rev | cut -d "/" -f1 | rev)
+
+if [[("$1" != "")]]; then
+   if [[("$BASE_DIRECTORY" != "$PROJECT_NAME")]]; then
+      echo "Make sure you are in $PROJECT_NAME project"
+      exit
+   fi
+   rootfileName=$1
+   mkdir ./$rootfileName
+   cp -a ./Python/ ./$rootfileName/
+   rm ./$rootfileName/*/*/*.py
+else
+   echo "Missing folder name: sh createFolders.sh <folder_name>"
```

Comment:
you can use the [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)

```
if [[ $# -lt 1 ]]; then
  echo "Usage: bash createFolders.sh <programming language>" > &2
  exit 1
fi
# The rest. You can now use $1 confidently.
```

the `>&2` prints to stderr.